File: Blueprints/BlueprintDefinitions/vs2022/SimpleKinesisFunction-FSharp/template/test/BlueprintBaseName.1.Tests/FunctionTest.fs
namespace BlueprintBaseName._1.Tests


open Xunit
open Amazon.Lambda.KinesisEvents
open Amazon.Lambda.TestUtilities

open System
open System.Collections.Generic
open System.IO
open System.Text

open BlueprintBaseName._1


module FunctionTest =
    [<Fact>]
    let ``Test Reading Kinesis Stream Event``() =
        let kinesisRecords = [
            KinesisEvent.KinesisEventRecord(
                EventId = "id-foo",
                EventName = "id-name",
                AwsRegion = "us-west-2",
                Kinesis = KinesisEvent.Record(
                    ApproximateArrivalTimestamp = DateTime.Now,
                    Data = new MemoryStream(Encoding.UTF8.GetBytes("Hello World Kinesis Record"))
                )
            )
        ]

        let kinesisEvent = KinesisEvent(Records = List(kinesisRecords))
        let lambdaFunction = Function()
        let testLogger = TestLambdaLogger()
        let context = TestLambdaContext(Logger = testLogger)
        lambdaFunction.FunctionHandler kinesisEvent context

        Assert.Contains("id-foo", testLogger.Buffer.ToString())
        Assert.Contains("Stream processing complete", testLogger.Buffer.ToString())

    [<EntryPoint>]
    let main _ = 0

File: Blueprints/BlueprintDefinitions/vs2022/SimpleSQSFunction/template/test/BlueprintBaseName.1.Tests/FunctionTest.cs
using Xunit;
using Amazon.Lambda.TestUtilities;
using Amazon.Lambda.SQSEvents;

namespace BlueprintBaseName._1.Tests;

public class FunctionTest
{
    [Fact]
    public async Task TestSQSEventLambdaFunction()
    {
        var sqsEvent = new SQSEvent
        {
            Records = new List<SQSEvent.SQSMessage>
            {
                new SQSEvent.SQSMessage
                {
                    Body = "foobar"
                }
            }
        };

        var logger = new TestLambdaLogger();
        var context = new TestLambdaContext
        {
            Logger = logger
        };

        var function = new Function();
        await function.FunctionHandler(sqsEvent, context);

        Assert.Contains("Processed message foobar", logger.Buffer.ToString());
    }
}